### PR TITLE
MDEV-16294 - INSTALL PLUGIN IF NOT EXISTS / UNINSTALL PLUGIN IF EXISTS

### PR DESCRIPTION
**Pull request by grooverdan**
Allow the installation / uninstallation of plugins without doing checks for existence first.

I submit this under the MCA.

### CODE REVIEW

**Comment by grooverdan** [comment]
rebased to use ```MYF(ME_NOTE)``` (ME_WARNING seem too harsh for something optional in the syntax)

File: mysql-test/main/plugin.test [hunk 1]
```diff
@@ -265,3 +265,22 @@ UNINSTALL PLUGIN example;
...
+--echo # UNINSTALL IF EXISTS PLUGIN|SONAME name
+--echo #
+--echo #
+
+select plugin_name from information_schema.plugins where plugin_library like 'ha_example%';
+INSTALL PLUGIN IF NOT EXISTS example SONAME 'ha_example';
+select plugin_name from information_schema.plugins where plugin_library like 'ha_example%';
+INSTALL PLUGIN IF NOT EXISTS EXAMPLE SONAME 'ha_example';
+SHOW WARNINGS;
+INSTALL SONAME IF NOT EXISTS 'ha_example';
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply mysql-test/main/plugin.test]: on 10.3 HEAD unmodified I tried this mtr test:
<pre>
--source include/not_windows_embedded.inc
--source include/have_example_plugin.inc

INSTALL PLUGIN example SONAME 'ha_example';
INSTALL SONAME 'ha_example';

SHOW PLUGINS;
<pre>

With the output:
<pre>
INSTALL PLUGIN example SONAME 'ha_example';
INSTALL SONAME 'ha_example';
SHOW PLUGINS;
Name	Status	Type	Library	License
...
EXAMPLE	ACTIVE	STORAGE ENGINE	ha_example.so	GPL
UNUSABLE	ACTIVE	DAEMON	ha_example.so	GPL
</pre>

<pre>
INSTALL SONAME 'ha_example';
INSTALL SONAME 'ha_example';
SHOW PLUGINS;
Name	Status	Type	Library	License
..
EXAMPLE	ACTIVE	STORAGE ENGINE	ha_example.so	GPL
UNUSABLE	ACTIVE	DAEMON	ha_example.so	GPL
</pre>

Is this intentional?

[the remark this answers is not in the document]
Comment (vuvova) [reply mysql-test/main/plugin.test]: To a degree, yes. INSTALL SONAME can cope with the case when there are many plugins in a given soname and some of them were already installed before INSTALL SONAME was executed.

Perhaps it should issue some kind of a warning if it'll find that **all** plugins were already installed. Or may be not.

**Comment by grooverdan** [comment]
Happy with this now?

**Comment by vuvova** [comment]
I usually merge pull requests as is (optionally with a follow-up commit of my own, if needed).
If you'd like me to squash the history, please say so.

**Comment by grooverdan** [comment]
either way is fine by me

**Comment by grooverdan** [comment]
thanks

**Pull request closed without merging**